### PR TITLE
feat: react blank-app-seed

### PR DESCRIPTION
**Pull request by ArturKrasinski**
[Review Guidance](https://www.notion.so/genesisglobal/Platform-Code-Review-Process-ace93ba760cc4563b0dfb712e2a88d8a)

🤔 &nbsp; **What does this PR do?**

- Adds the option to choose a UI framework: React.

🚀 &nbsp; **Where should the reviewer start?**

- files in: .genx/templates/react 
- files in: client-tmp/react

📑 &nbsp; **How should this be tested?**

### Route and CSV parameter handling test (react)

```
rm -rf blankappseedtest && npx -y @genesislcap/genx@latest init blankappseedtest --ref ak/react-example --no-npm -x --routes '[{"name":"Home","tiles":[{"title":"Entity manager","type":"entity-manager","config":{ "modalPosition": "centre", "sizeColumnsToFit": true, "enableSearchBar": true, "resourceName":"ALL_POSITIONS","title":"My Positions","updateEvent":"EVENT_COUNTERPARTY_MODIFY","deleteEvent":"EVENT_COUNTERPARTY_DELETE","createEvent":"EVENT_COUNTERPARTY_INSERT", "createFormUiSchema": {"type":"VerticalLayout","elements":[{"type":"Control","label":"Inline create schema - main contact","scope":"#/properties/MAIN_CONTACT"},{"type":"Control","label":"Issuer Name - Local Schema","scope":"#/properties/ISSUER_NAME","options":{"readonly":true}},{"type":"Control","label":"Price","scope":"#/properties/PRICE"},{"type":"Control","scope":"#/properties/COUNTERPARTY","options":{"allOptionsResourceName":"ALL_COUNTERPARTYS","valueField":"COUNTERPARTY_ID","labelField":"COUNTERPARTY_ID"}},{"type":"Control","label":"Password","scope":"#/properties/PASSWORD","options":{"isPassword":true}},{"type":"Control","label":"Password","scope":"#/properties/PASSWORD","options":{"textarea":true}}]}, "updateFormUiSchema": {"type":"VerticalLayout","elements":[{"type":"Control","label":"Inline update schema - main contact","scope":"#/properties/MAIN_CONTACT"},{"type":"Control","label":"Issuer Name - Local Schema","scope":"#/properties/ISSUER_NAME","options":{"readonly":true}},{"type":"Control","label":"Price","scope":"#/properties/PRICE"},{"type":"Control","scope":"#/properties/COUNTERPARTY","options":{"allOptionsResourceName":"ALL_COUNTERPARTYS","valueField":"COUNTERPARTY_ID","labelField":"COUNTERPARTY_ID"}},{"type":"Control","label":"Password","scope":"#/properties/PASSWORD","options":{"isPassword":true}},{"type":"Control","label":"Password","scope":"#/properties/PASSWORD","options":{"textarea":true}}]}, "columns": [{"field":"INSTRUMENT_NAME","headerName":"Instrument Name"},{"field":"VALUE","headerName":"Inline coldef - VALUE"},{"field":"QUANTITY","headerName":"Quantity"},{"field":"PNL","headerName":"PNL"}] }},{"title":"Grid","type":"grid-pro","config":{"resourceName":"ALL_TRADES", "deferredGridOptions": {"columnDefs":[{"field":"INSTRUMENT_NAME","headerName":"Instrument Name"},{"field":"VALUE","headerName":"Inline coldef - VALUE"},{"field":"QUANTITY","headerName":"Quantity"},{"field":"PNL","headerName":"PNL"}]} }},{"title":"Form","type":"smart-form","config":{"resourceName":"EVENT_COUNTERPARTY_INSERT", "uischema": {"type":"VerticalLayout","elements":[{"type":"Control","label":"Inline form schema - main contact","scope":"#/properties/MAIN_CONTACT"},{"type":"Control","label":"Issuer Name - Local Schema","scope":"#/properties/ISSUER_NAME","options":{"readonly":true}},{"type":"Control","label":"Price","scope":"#/properties/PRICE"},{"type":"Control","scope":"#/properties/COUNTERPARTY","options":{"allOptionsResourceName":"ALL_COUNTERPARTYS","valueField":"COUNTERPARTY_ID","labelField":"COUNTERPARTY_ID"}},{"type":"Control","label":"Password","scope":"#/properties/PASSWORD","options":{"isPassword":true}},{"type":"Control","label":"Password","scope":"#/properties/PASSWORD","options":{"textarea":true}}]} }}]}, {"name":"Realtime Dashboard","tiles":[{"title":"Entity manager tile","type":"entity-manager","config":{"resourceName":"ALL_COUNTERPARTYS","title":"Counterparty Management","updateEvent":"EVENT_COUNTERPARTY_MODIFY","deleteEvent":"EVENT_COUNTERPARTY_DELETE","createEvent":"EVENT_COUNTERPARTY_INSERT"}},{"title":"Form tile","type":"smart-form","config":{"resourceName":"EVENT_COUNTERPARTY_INSERT"}}]},{"name":"Analytics","tiles":[{"title":"Grid Tile","type":"grid-pro","config":{"resourceName":"ALL_POSITIONS"}},{"title":"Charts Tile 1","type":"chart","config":{"type":"line","resourceName":"ALL_POSITIONS","xField":"INSTRUMENT_NAME","yField":"VALUE"}},{"title":"Charts Tile 2","type":"chart","config":{"type":"pie","resourceName":"ALL_POSITIONS","xField":"INSTRUMENT_NAME","yField":"VALUE"}},{"title":"Charts Tile 3","type":"chart","config":{"type":"column","resourceName":"ALL_POSITIONS","xField":"INSTRUMENT_ID","yField":"VALUE"}}]}]' --csv '[{"name": "trade", "fields": ["a", "B"]}, {"name": "position", "fields": ["id", "TYPE"]} ]' --apiHost 'wss://public-foundation.genesislab.global/gwf/' --no-shell --framework React && cd blankappseedtest/client && npm run bootstrap && npm run dev
```

✅ &nbsp; **Checklist**

- [x] I have tested my changes.
- [ ] I have added tests for my changes.
- [ ] I have updated the project documentation to reflect my changes.

### CODE REVIEW

File: .genx/utils/generateRoute.js [hunk 1]
```diff
@@ -4,8 +4,10 @@ const generateTile = require('./generateTile');
 const {
   FRAMEWORK_WEB_COMPONENTS_ALIAS,
   FRAMEWORK_ANGULAR_ALIAS,
+  FRAMEWORK_REACT_ALIAS,
   DIR_TEMPLATE_BY_FRAMEWORK,
 } = require('../static');
+const { get } = require('node:http');
```
Comment (cistov) [inline .genx/utils/generateRoute.js:10]: Unused?

File: client-tmp/react/src/config.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  '/auth': 'blank',
+  '/': 'blank',
+};
+
+import type { MainMenu } from './types/menu';
+
+export const AUTH_PATH = 'auth';
+
+export const API_DATA = {
+  URL: 'wss://public-foundation.genesislab.global/gwf/',
```
Comment (cistov) [inline client-tmp/react/src/config.ts:14]: hard-coded just for testing I'm guessing

File: client-tmp/react/README.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
+
+# GenesisLCAP React Showcase Client Application
```
Comment (cistov) [inline client-tmp/react/README.md]: Copied from integration repo I'm guessing :) Have a look at the Web components readme

File: client-tmp/react/src/config.ts (hunk 2 above)
Comment (ArturKrasinski) [reply client-tmp/react/src/config.ts:14]: yup, it's not ready to merge / review

File: client-tmp/react/.gitignore [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+# misc
+.DS_Store
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
```
Comment (cistov) [inline client-tmp/react/.gitignore:23]: Other than pnp everything should already be covered by the root .gitignore. Not-sure if we're actually using that pnp stuff?

Comment (ArturKrasinski) [reply client-tmp/react/.gitignore]: removed

File: client-tmp/react/assets/react.svg
[the remark this answers is not in the document]
Comment (ArturKrasinski) [reply client-tmp/react/assets/react.svg:1]: Right, it was created when using Create React App - I removed it.

File: client-tmp/react/README.md (hunk 3 above)
Comment (ArturKrasinski) [reply client-tmp/react/README.md]: I added the basic information + npm scripts to be consistent with those from the web components.

**Review by skawian — APPROVED** [review]

**Review by MrBrunoWolff — APPROVED** [review]

File: client-tmp/react/src/assets/logo-icon--light.svg
Comment (MrBrunoWolff) [inline client-tmp/react/src/assets/logo-icon--light.svg:1]: nit: curious if we really need these .svg files!

File: client-tmp/react/src/assets/react-icon.svg
Comment (MrBrunoWolff) [inline client-tmp/react/src/assets/react-icon.svg:1]: same here!

File: client-tmp/react/src/assets/logo-icon--light.svg
Comment (skawian) [reply client-tmp/react/src/assets/logo-icon--light.svg:1]: i think this one is genesis one

File: client-tmp/react/src/assets/react-icon.svg
Comment (skawian) [reply client-tmp/react/src/assets/react-icon.svg:1]: this probably not needed

Comment (skawian) [reply client-tmp/react/src/assets/react-icon.svg:1]: removed

**Pull request merged**